alacritty/src/window_context.rs

```diff
@@ -391,7 +391,11 @@ impl WindowContext {
 
         // Request immediate re-draw if visual bell animation is not finished yet.
         if !self.display.visual_bell.completed() {
-            self.display.window.request_redraw();
+            let window_id = Some(self.display.window.id());
+            // Winit is not reliable on at least Windows when asking for `Redraw` during the
+            // `RedrawRequsted`.
```

Comment:
@chrisduerr it's a windows itself, as I linked. Winit will need to change how it works entirely and not rely on `WM_PAINT` for drawing.

>Also we shouldn't just be abusing other events just because they might work. If we need our own event, we should define it properly.

They not **might**, they work. Because they are user events. Windows itself preventing winit from requesting a redraw without any signs.